engine/src/conversion/codegen_cpp/mod.rs

```diff
@@ -432,6 +432,11 @@ impl<'a> CppCodeGenerator<'a> {
                 CppFunctionBody::Destructor(ns, id) => {
                     let ty_id = QualifiedName::new(ns, id.clone());
                     let ty_id = identifier_using_original_name_map(&ty_id, &self.original_name_map);
+                    // If the name refers to a nested identifier, then we only want the last part
+                    // of it to name the destructor.
```

Comment:
Would you mind pulling this bit and its associated test out into a separate PR? (I think I'm right in saying that it's independent from the rest of these improvements?)

I'm surprised that you're having to do the `rsplit` here, and I reckon this is probably best done by improving the way the rest of autocxx tracks the name of nested types. I don't want to delay the rest of this fine PR while I figure out what I did wrong here...

Reply:
Separate, but necessary for some of the existing tests to pass because they get destructors now. Pulled out to #864, but I'm going to leave the diff here so tests pass. Should merge without conflicts if there are no further changes, or I'll resolve them in this PR if the other one merges first.